### PR TITLE
Add `/userinfo`

### PR DESCRIPTION
**Pull request by Djacon**
The command now shows main information about the Codewars user

### CODE REVIEW

File: src/commands/userinfo.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+${authored ? `\nCreated katas: ${authored}`: ''}
+Completed katas: ${completed}
+\`\`\``
+};
+
+export const data = async () =>
+  new SlashCommandBuilder()
+    .setName("userinfo")
+    .setDescription("Info about a user")
+    .addStringOption((opt) => opt.setName("username").setDescription("The Codewars username for information about the user"))
```
Comment (kazk) [inline src/commands/userinfo.ts:33]:
```suggestion
    .addStringOption((opt) => opt.setName("username").setDescription("The username to look up"))
```

File: src/commands/userinfo.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    .addBooleanOption((opt) => opt.setName("ephemeral").setDescription("Don't show user statistics to others"))
```
Comment (kazk) [inline src/commands/userinfo.ts:34]:
```suggestion
    .addBooleanOption((opt) => opt.setName("ephemeral").setDescription("Hide from others"))
```

File: src/commands/userinfo.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      ephemeral: true,
+    });
+    return;
+  }
+  
+  await interaction.reply({
+    content: content,
+    ephemeral: ephemeral,
+  });
+  return;
```
Comment (kazk) [inline src/commands/userinfo.ts:67]:
```suggestion
```

**Review by kazk — CHANGES_REQUESTED** [review]
Looks good overall. Left some minor change requests.

File: src/commands/userinfo.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  const authored = user.codeChallenges.totalAuthored;
+  const completed = user.codeChallenges.totalCompleted;
+
+  return `\`\`\`
+User: ${name}
+Rank: ${rank}
+Honor: ${honor}\
+${position ? `\nPosition: #${position}`: ''}\
+${authored ? `\nCreated katas: ${authored}`: ''}
+Completed katas: ${completed}
```
[the remark this answers is not in the document]
Comment (Djacon) [reply src/commands/userinfo.ts:25]: Hmm, first time I've heard of it, ok, I'll know.

**Comment by Djacon** [comment]
kazk, how to accept a request for PR changes?

**Comment by kazk** [comment]
What do you mean by accept a request? You can push the changes.

For the lines I added suggestions, you can use the button to commit suggestion. Go to https://github.com/codewars/discord-bot/pull/69/files and you should see something similar to:

image

Use "Add suggestion to batch" to commit multiple suggestions together.

File: src/commands/userinfo.ts (hunk 4 above)
Comment (kazk) [inline src/commands/userinfo.ts:25]:
```suggestion
${authored ? `\nCreated kata: ${authored}`: ''}
Completed kata: ${completed}
```

**Comment by Djacon** [comment]
> Use "Add suggestion to batch" to commit multiple suggestions together.

Yes, I meant this one, thanks

**Review by kazk — APPROVED** [review]

**Pull request merged**